src/slider/utils.ts

```diff
@@ -0,0 +1,24 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+export const getPercent = (value: number, min: number, max: number) => ((value - min) / (max - min)) * 100;
+
+export const getStepArray = (step: number, min: number, max: number) => {
+  const steps = [];
+
+  for (let i = min; i <= max; i = i + step) {
+    steps.push(i);
+  }
+  return steps;
+};
+
+export const findLowerAndHigherVals = (array: Array<number>, value: number) => {
+  array.sort((a, b) => a - b);
+
+  const index = array.indexOf(value);
+  const lower = array[index - 1];
+  const higher = array[index + 1];
+
+  return { lower, higher };
```

Comment:
When using index access like `array[index - 1]` TypeScript (with our settings) assumes the value is always present, yet it might be undefined. You can add the return type to the function `{ lower: undefined | number, higher: undefined | number }` to express the real type. Otherwise, the consumer code might rely on the values being always defined.